execution/gethexec/caff_node.go

```diff
@@ -189,9 +228,67 @@ func (n *CaffNode) Start(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("failed to start node, error in createBlock: %w", err)
 	}
+
+	err = n.CallIterativelySafe(func(ctx context.Context) time.Duration {
```

Comment:
Why is this like an iterative call? Why cant we just write to the database when we update the nextHotshotBlockNum